lib/api/model/model.dart

```diff
@@ -604,6 +613,37 @@ extension type const TopicName(String _value) {
   /// using [canonicalize].
   bool isSameAs(TopicName other) => canonicalize() == other.canonicalize();
 
+  /// Convert this topic to match how it would appear on a message object from
+  /// the server.
+  ///
+  /// This assumes that the topic is originally for a send-message
+  /// request — such a topic must be constructed from a string without
+  /// leading/trailing whitespace.
```

Comment:
Hmm, is that a requirement of the server endpoint? I don't see it here:
https://zulip.com/api/send-message#parameter-topic

Fine for this method to have this requirement; just shouldn't attribute it to the server.